File: ChallengeAlpha/app/src/main/java/com/hurb/challengealpha/model/Address.kt
package com.hurb.challengealpha.model

import com.google.gson.annotations.SerializedName

data class Address(

    @SerializedName("zipcode") val zipcode: String,
    @SerializedName("street") val street: String,
    @SerializedName("street_name") val street_name: String,
    @SerializedName("streetName") val streetName: String,
    @SerializedName("address") val address: String,
    @SerializedName("fullAddress") val fullAddress: String,
    @SerializedName("full_address") val full_address: String,
    @SerializedName("neighborhood") val neighborhood: String,
    @SerializedName("id_atlas_neighborhood") val id_atlas_neighborhood: String,
    @SerializedName("id_neighborhood") val id_neighborhood: String,
    @SerializedName("city") val city: String,
    @SerializedName("id_atlas_city") val id_atlas_city: String,
    @SerializedName("id_city") val id_city: Int,
    @SerializedName("state") val state: String,
    @SerializedName("id_atlas_state") val id_atlas_state: String,
    @SerializedName("id_state") val id_state: Int,
    @SerializedName("country") val country: String,
    @SerializedName("id_atlas_country") val id_atlas_country: String,
    @SerializedName("id_country") val id_country: Int,
    @SerializedName("countryAlfa2") val countryAlfa2: String,
    @SerializedName("country_alfa2") val country_alfa2: String,
    @SerializedName("geoLocation") val geoLocation: GeoLocation
)

File: DesafioHurb/Domain/Model/Address.swift
//
//  Address.swift
//  DesafioHurb
//
//  Created by Edson Aparecido Guido on 13/09/20.
//  Copyright © 2020 Edson Aparecido Guido. All rights reserved.
//

import Foundation

struct Address: Codable {
    let zipcode: String?
    let addressFullAddress: String?
    let street: String?
    let addressStreetName: String?
    let streetName: String?
    let address: String?
    let fullAddress: String?
    let neighborhood: String?
    let idAtlasNeighborhood: String?
    let idNeighborhood: Int?
    let city: String
    let idAtlasCity: Int?
    let idCity: Int?
    let state: String
    let idAtlasState: Int?
    let idState: Int?
    let country: String
    let idAtlasCountry: Int?
    let idCountry: Int?
    let geoLocation: GeoLocation

    enum CodingKeys: String, CodingKey {
        case zipcode
        case addressFullAddress = "full_address"
        case street
        case addressStreetName = "street_name"
        case streetName, address, fullAddress, neighborhood
        case idAtlasNeighborhood = "id_atlas_neighborhood"
        case idNeighborhood = "id_neighborhood"
        case city
        case idAtlasCity = "id_atlas_city"
        case idCity = "id_city"
        case state
        case idAtlasState = "id_atlas_state"
        case idState = "id_state"
        case country
        case idAtlasCountry = "id_atlas_country"
        case idCountry = "id_country"
        case geoLocation
    }
}
